### PR TITLE
Support navigation/references for init functions

### LINKED ISSUE #42985: [Improvement]: VS Code Code Navigation Does not Work with `init` Functions
### Description

In the VS Code, when we try to use code navigation on an `init` function of an object, it does not show any usage. Ideally, this should show the places where the object is initialized (called `new`).

### Describe your problem(s)

_No response_

### Describe your solution(s)

_No response_

### Related area

-> Compilation

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by LakshanWeerasinghe**
## Purpose
$subject

Fixes #42985

## Approach
In the ReferenceFinder previously we weren't visiting the init function invocations.

## Samples
https://github.com/user-attachments/assets/2e5a7abd-fd39-4ba3-89cf-d0e228d050fe

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by dulajdilshan — COMMENTED** [review]
Does this cover the other way too (Clicking the new keyword and goes to the init function)?

File: compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/ReferenceFinder.java [hunk 1]
```diff
@@ -898,7 +898,7 @@ public void visit(BLangInvocation invocationExpr) {
     @Override
     public void visit(BLangTypeInit typeInit) {
         find(typeInit.userDefinedType);
-        find(typeInit.argsExpr);
```
[the remark this answers is not in the document]
Comment (LakshanWeerasinghe) [reply compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/ReferenceFinder.java:901]: args will be visited when we visit the initInvocation node.

**Comment by LakshanWeerasinghe** [comment]
> Does this cover the other way too (Clicking the new keyword and goes to the init function)?

No. If this is explicit invocation when we click the class descriptor it will navigate to the class definition. And also new is not a symbol, its a key word. When thinking of that navigating to the init function also not correct.

**Comment by dulajdilshan** [comment]
> > Does this cover the other way too (Clicking the new keyword and goes to the init function)?
> 
> No. If this is explicit invocation when we click the class descriptor it will navigate to the class definition. And also new is not a symbol, its a key word. When thinking of that navigating to the init function also not correct.

Even though `new` is a keyword, we point `init` to a `new`, therefore the other way round is also okay. But your logic is valid.

File: compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/ReferenceFinder.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (dulajdilshan) [reply compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/ReferenceFinder.java:901]: Ack.

**Review by dulajdilshan — APPROVED** [review]

**Pull request merged**